### PR TITLE
Make edl controller jobparser workable

### LINKED ISSUE #6: EDL controller does not submit created resources currently
We intend to submit ReplicaSets and Job for each `TrainingJob` this is not implemented yet, 
see code from here: https://github.com/PaddlePaddle/edl/blob/develop/pkg/controller.go#L133

### PR DESCRIPTION
**Pull request by typhoonzero**
Fix #6 and add some examples.

### CODE REVIEW

File: example/examplejob.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+apiVersion: paddlepaddle.org/v1
+kind: TrainingJob
+metadata:
+  name: example
+spec:
+  #image: "paddlepaddle/paddlecloud-job"
+  image: "typhoon1986/paddlecloud-job:wuyitest"
```
Comment (Yancey0623) [inline example/examplejob.yaml]: Maybe we can keep the official Docker image.

Comment (typhoonzero) [reply example/examplejob.yaml]: Done

File: edl_controller.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: training-job-controller
+  namespace: paddlecloud
```
Comment (putcn) [inline edl_controller.yaml:5]: do we need to create the namespace before this yaml can claim?

**Comment by putcn** [comment]
excellent! I was trying the same yesterday to make the whole thing work. also, to make the training job work as CRD, do we just need to change the yaml(thirdpartyresource.yaml) or there are go code need to change with that?

File: Dockerfile [hunk 3]
```diff
@@ -2,8 +2,14 @@ FROM    golang:1.8
...
 WORKDIR $GOPATH/src/github.com/paddlepaddle
-RUN     git clone https://github.com/paddlepaddle/edl.git
+RUN     mkdir -p $GOPATH/src/github.com/paddlepaddle/edl
+# Add ENV http_proxy=[your proxy server] if needed
+# run glide install before building go sources, so that
+# if we change the code and rebuild the image can cost little time
+ADD     ./glide.yaml ./glide.lock $GOPATH/src/github.com/paddlepaddle/edl/
 WORKDIR $GOPATH/src/github.com/paddlepaddle/edl
 RUN     glide install --strip-vendor
+ADD     . $GOPATH/src/github.com/paddlepaddle/edl
```
Comment (helinwang) [inline Dockerfile:12]: Thanks! This is the correct solution.

**Review by helinwang — DISMISSED** [review]
Thank you! LGTM++

File: edl_controller.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        name: training-job-controller
+    spec:
+      containers:
+      - name: training-job-controller
+        image: paddlepaddle/edl-controller
+        imagePullPolicy: Never
```
Comment (putcn) [inline edl_controller.yaml]: should we change the policy to `IfNotPresent`?

File: edl_controller.yaml (hunk 2 above)
Comment (typhoonzero) [reply edl_controller.yaml:5]: Indeed. We should update the documents later.

**Comment by typhoonzero** [comment]
>excellent! I was trying the same yesterday to make the whole thing work. also, to make the training job work as CRD, do we just need to change the yaml(thirdpartyresource.yaml) or there are go code need to change with that?

@putcn Yes, I'm afraid we have to put the CRD implement in another branch, it should have a lot of difference from this one.

File: edl_controller.yaml (hunk 4 above)
Comment (typhoonzero) [reply edl_controller.yaml]: Done.

**Review by putcn — APPROVED** [review]
LGTM!

**Pull request merged**